ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/categories/AppCategoriesPresenterImpl.java

```diff
@@ -84,201 +82,150 @@ public int compare(AppCategory group1, AppCategory group2) {
-                    }
-
-                    @Override
-                    public void onSuccess(final App result) {
-                        // Create list of group hierarchies
-                        List<List<String>> appGroupHierarchies = Lists.newArrayList();
-                        for (AppCategory appCategory : result.getGroups()) {
-                            appGroupHierarchies.add(getGroupHierarchy(appCategory));
-                        }
-
-                        dlg.show(result,
-                                 searchRegexPattern,
-                                 appGroupHierarchies,
-                                 AppCategoriesPresenterImpl.this,
-                                 AppCategoriesPresenterImpl.this,
-                                 AppCategoriesPresenterImpl.this);
-                    }
-                });
-            }
-        });
+    void createHPCTab() {
+        hpcView.addAppCategorySelectedEventHandler(this);
+        addAppCategories(hpcView.getTree().getStore(), null, hpcCategories);
     }
 
-    @Override
-    public void onAppRatingDeselected(AppRatingDeselected event) {
-        final App appToUnRate = event.getApp();
-        appUserService.deleteRating(appToUnRate, new DeleteRatingCallback(appToUnRate,
-                                                                      eventBus));
+    void createWorkspaceTab(HasId selectedAppCategory) {
+        TreeStore<AppCategory> treeStore = workspaceView.getTree().getStore();
+        final Store.StoreSortInfo<AppCategory> info = new Store.StoreSortInfo<>(new AppCategoryComparator(treeStore),
+                                                                                SortDir.ASC);
+        treeStore.addSortInfo(info);
+        workspaceView.addAppCategorySelectedEventHandler(this);
+        addAppCategories(treeStore, null, workspaceCategories);
+        if (selectedAppCategory != null) {
+            AppCategory desiredCategory = treeStore.findModelWithKey(selectedAppCategory.getId());
+            workspaceView.getTree().getSelectionModel().select(desiredCategory, false);
+        } else {
+            workspaceView.getTree().getSelectionModel().selectNext();
+            workspaceView.getTree().getSelectionModel().select(treeStore.getRootItems().get(0), false);
+        }
     }
 
-    @Override
-    public void onAppRatingSelected(AppRatingSelected event) {
-        final App appToRate = event.getApp();
-        appUserService.rateApp(appToRate,
-                               event.getScore(),
-                               new RateAppCallback(appToRate,
-                                                   eventBus));
+    void filterCategories(List<AppCategory> result) {
+        workspaceCategories.clear();
+        hpcCategories.clear();
+        for (AppCategory category : result) {
+            if (!category.isPublic()) {
```

Comment:
Rather than filtering public categories from the response, the endpoint supports a `public` flag. If set to `false`, it will only return the user's private workspace categories.

It currently includes the public HPC category in the `public=false` response as well. When that logical error gets fixed in the future, then maybe the UI could just make the HPC category itself, instead of fetching it from the server, since its ID is now stored in the UI configs.